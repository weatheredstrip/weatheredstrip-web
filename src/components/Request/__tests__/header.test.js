import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Header from '../header.react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import 'jest-dom/extend-expect'
afterEach(cleanup)

function renderWithRouter(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  }
}

it('header renders properly', () => {
  const { container } = renderWithRouter(<Header />)
  const header = container.firstChild
  expect(header).toMatchSnapshot()
})
