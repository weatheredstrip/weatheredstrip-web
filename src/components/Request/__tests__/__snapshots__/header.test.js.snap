// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders properly 1`] = `
<header
  class="app-header"
>
  <div
    class="app-header-content"
  >
    <a
      href="/"
    >
      <div
        id="app-info"
      >
        <img
          alt="App Logo"
          id="app-logo"
          src="logo.svg"
        />
        <div
          class="app-title"
        >
          Weathered Strip
        </div>
      </div>
    </a>
    <div
      class="search-nav css-nil ðŸ“¦box-szg_border-box"
    >
      <form
        class="searchbox"
      >
        <input
          aria-label="search-input"
          class="searchbox-input"
          placeholder="ICAO or IATA codes..."
          type="text"
          value=""
        />
        <button
          aria-label="search-button"
          class="searchbox-button"
          disabled=""
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
            />
          </svg>
        </button>
      </form>
      <div
        class="css-nil ðŸ“¦pst_relative ðŸ“¦box-szg_border-box"
      >
        <button
          class="css-1ii3p2c ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_0px ðŸ“¦pl_0px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦w_32px ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box"
        >
          <svg
            class="ðŸ“¦w_24px ðŸ“¦h_24px ðŸ“¦box-szg_border-box"
            data-icon="menu"
            style="fill: #66788A;"
            viewBox="0 0 20 20"
          >
            <path
              d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</header>
`;
